pkg/cloudprovider/providers/vsphere/vclib/virtualmachine.go

```diff
@@ -400,3 +401,10 @@ func (vm *VirtualMachine) deleteController(ctx context.Context, controllerDevice
 	}
 	return nil
 }
+
+// RefreshThisVM refreshes this virtual machine with new client connection. Caller needs to take care of concurrency by using appropriate locks.
+func (vm *VirtualMachine) RenewVM(client *govmomi.Client) VirtualMachine {
```

Comment:
Since this function is not thread safe, shouldn't we add some synchronization in the caller above - renewNodeInfo?

Reply:
renewNodeInfo is side-effects free. So no synchronization needed. GetNodeInfo and GetNodeDetails have appropriate synchronization in place. Let me know if you still think that there is a possibility of race condition.